### PR TITLE
Add AITool -> OpenAI.Responses.ResponseTool conversion utility

### PR DESCRIPTION
**Pull request by rogerbarreto**
This pull request adds support for converting a wider range of AI tool types into OpenAI `ResponseTool` instances, improving interoperability and extensibility. The main update is the introduction of a new conversion method and its comprehensive test coverage, ensuring correct behavior for all supported tool types and edge cases.

### Conversion method enhancements

* Added a new extension method `AsOpenAIResponseTool` to the `MicrosoftExtensionsAIResponsesExtensions` class, allowing conversion from any `AITool` to an OpenAI `ResponseTool`, or returning `null` if unsupported.
* Implemented the core conversion logic in `OpenAIResponsesChatClient.ToResponseTool`, supporting multiple tool types including `AIFunctionDeclaration`, `HostedWebSearchTool`, `HostedFileSearchTool`, `HostedCodeInterpreterTool`, and `HostedMcpServerTool`, with detailed handling for each type's properties and options.

### Testing improvements

* Added thorough unit tests for the new conversion method, covering all supported tool types, property mapping, edge cases (unknown tool types, null input), and ensuring correct output types and values.
* Introduced a helper class `UnknownAITool` in the test suite to verify behavior when an unsupported tool type is passed to the conversion method.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/6958)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs [hunk 1]
```diff
@@ -492,96 +591,10 @@ private ResponseCreationOptions ToOpenAIResponseCreationOptions(ChatOptions? opt
         {
             foreach (AITool tool in tools)
             {
-                switch (tool)
+                var responseTool = ToResponseTool(tool, options);
+                if (responseTool is not null)
```
Comment (stephentoub) [inline src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs]:
```suggestion
                if (ToResponseTool(tool, options) is { } responseTool)
```

File: src/Libraries/Microsoft.Extensions.AI.OpenAI/MicrosoftExtensionsAIResponsesExtensions.cs [hunk 2]
```diff
@@ -21,6 +21,13 @@ public static class MicrosoftExtensionsAIResponsesExtensions
     public static FunctionTool AsOpenAIResponseTool(this AIFunctionDeclaration function) =>
         OpenAIResponsesChatClient.ToResponseTool(Throw.IfNull(function));
 
+    /// <summary>Creates an OpenAI <see cref="ResponseTool"/> from an <see cref="AITool"/>.</summary>
+    /// <param name="tool">The tool to convert.</param>
+    /// <returns>An OpenAI <see cref="ResponseTool"/> representing <paramref name="tool"/> or <see langword="null"/> if there is no mapping.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="tool"/> is <see langword="null"/>.</exception>
```
Comment (stephentoub) [inline src/Libraries/Microsoft.Extensions.AI.OpenAI/MicrosoftExtensionsAIResponsesExtensions.cs:27]:
```suggestion
    /// <exception cref="ArgumentNullException"><paramref name="tool"/> is <see langword="null"/>.</exception>
    /// <remarks>
    /// This method is only able to create <see cref="ResponseTool"/>s for <see cref="AITool"/> types
    /// it's aware of, namely all of those available from the Microsoft.Extensions.AI.Abstractions library.
    /// </remarks>
```

File: test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAIConversionTests.cs [hunk 3]
```diff
@@ -112,6 +112,168 @@ public void AsOpenAIResponseTool_ProducesValidInstance()
         Assert.NotNull(tool);
     }
 
+    [Fact]
+    public void AsOpenAIResponseTool_WithAIFunctionDeclaration_ProducesValidFunctionTool()
+    {
+        var tool = _testFunction.AsOpenAIResponseTool();
+
+        Assert.NotNull(tool);
+        Assert.IsType<FunctionTool>(tool);
```
Comment (stephentoub) [inline test/Libraries/Microsoft.Extensions.AI.OpenAI.Tests/OpenAIConversionTests.cs]: These tests should ideally validate properties on the returned tool, ensuring that things propagated as expected.

File: src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs [hunk 4]
```diff
@@ -431,6 +431,105 @@ void IDisposable.Dispose()
...
+            case AIFunctionDeclaration aiFunction:
+                bool? strict =
+                            OpenAIClientExtensions.HasStrict(aiFunction.AdditionalProperties) ??
+                            OpenAIClientExtensions.HasStrict(options?.AdditionalProperties);
+
+                return ResponseTool.CreateFunctionTool(
+                    aiFunction.Name,
+                    OpenAIClientExtensions.ToOpenAIFunctionParameters(aiFunction, strict),
+                    strict,
+                    aiFunction.Description);
```
Comment (stephentoub) [inline src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs]:
```suggestion
                return ToResponseTool(aiFunction, options);
```

File: src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs [hunk 5]
```diff
@@ -431,6 +431,105 @@ void IDisposable.Dispose()
...
+            case HostedFileSearchTool fileSearchTool:
+                return ResponseTool.CreateFileSearchTool(
+                    fileSearchTool.Inputs?.OfType<HostedVectorStoreContent>().Select(c => c.VectorStoreId) ?? [],
+                    fileSearchTool.MaximumResultCount);
+
+            case HostedCodeInterpreterTool codeTool:
+                return ResponseTool.CreateCodeInterpreterTool(
+                        new CodeInterpreterToolContainer(codeTool.Inputs?.OfType<HostedFileContent>().Select(f => f.FileId).ToList() is { Count: > 0 } ids ?
+                            CodeInterpreterToolContainerConfiguration.CreateAutomaticContainerConfiguration(ids) :
+                            new()));
```
Comment (stephentoub) [inline src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs]:
```suggestion
                    new CodeInterpreterToolContainer(codeTool.Inputs?.OfType<HostedFileContent>().Select(f => f.FileId).ToList() is { Count: > 0 } ids ?
                        CodeInterpreterToolContainerConfiguration.CreateAutomaticContainerConfiguration(ids) :
                        new()));
```

File: src/Libraries/Microsoft.Extensions.AI.OpenAI/MicrosoftExtensionsAIResponsesExtensions.cs [hunk 6]
```diff
@@ -21,6 +21,17 @@ public static class MicrosoftExtensionsAIResponsesExtensions
...
+    /// <remarks>
+    /// This method is only able to create <see cref="ResponseTool"/>s for <see cref="AITool"/> types
+    /// it's aware of, namely all of those available from the Microsoft.Extensions.AI.Abstractions library.
+    /// </remarks>
+    public static ResponseTool? AsOpenAIResponseTool(this AITool tool)
+        => OpenAIResponsesChatClient.ToResponseTool(Throw.IfNull(tool));
```
Comment (stephentoub) [inline src/Libraries/Microsoft.Extensions.AI.OpenAI/MicrosoftExtensionsAIResponsesExtensions.cs]:
```suggestion
    public static ResponseTool? AsOpenAIResponseTool(this AITool tool) =>
        OpenAIResponsesChatClient.ToResponseTool(Throw.IfNull(tool));
```
